homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -16,6 +16,8 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     async def async_step_import(self, import_config=None):
         """Handle the initial step."""
-        await self.async_set_unique_id(DOMAIN)
-        self._abort_if_unique_id_configured(import_config)
+        entry = await self.async_set_unique_id(DOMAIN)
+        if entry and import_config.items() != entry.data.items():
+            self.hass.config_entries.async_update_entry(entry, data=import_config)
+            return self.async_abort(reason="already_configured")
```

Comment:
The _abort_if_unique_id_configured only appends root keys. It won't update root keys. We want full replace here.

Reply:
Shouldn't the update entry be awaited here? The update must be done before the setup entry runs.

Reply:
Hmm. I don't think it's async def. Think it's @callback

Reply:
Yeah, you're right

Reply:
What do you mean with "only appends root keys"?

Reply:
It does:
{**olddict, **newdict} 

That mean anything missing in newdict at the root of the dict (ie not sub dicts) will be kept from olddict.

Reply:
Overlapping keys between olddict and newdict will be updated by newdict.

Reply:
Yup, but keys will not be removed.

Reply:
That is true. So we want keys that are not present in newdict to be removed from olddict when we update the config entry. Ok.